### PR TITLE
Instant Search: Polish search interface UX

### PR DESCRIPTION
**Pull request by jsnmoon**
Please note that the commit log includes previously merged changes for Instant Search. The new code introduced by this PR is from Feb 12, 2020, and onwards.

#### Changes proposed in this Pull Request:
* This PR primarily focuses on improving the overall user experience for Jetpack Instant Search.

>Before:
>
>Screen Shot 2020-02-21 at 3 42 30 PM

>After:
>
>Screen Shot 2020-02-21 at 3 40 31 PM

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* No, this is an incremental update for Jetpack Instant Search.

#### Testing instructions:
1. Follow [these setup instructions](https://github.com/Automattic/jetpack/blob/instant-search-master/modules/search/instant-search/README.md#testing-instructions) to set up Jetpack Instant Search on your site. Please note that the Jetpack Search widget will need to be added to twice: once to your site sidebar/footer and once to the Jetpack Search Sidebar within the search overlay.

2. Submit a query into your site search input. Ensure that a search overlay appears with your search results.

3. Try using the new sort select controls below the overlay search input. Ensure that they behave as expected.

4. Try resizing your window to be narrower than 768 pixels; a Filters button should spawn adjacent to the sort select controls.

5. Try clicking on the Filters button in (4). Ensure that a filters popover spawns below the button, like so:
Screen Shot 2020-02-21 at 3 47 34 PM

#### Proposed changelog entry for your changes:
* Interface refinements and bugfixes for Jetpack Instant Search.

### CODE REVIEW

**Review by gibrown — APPROVED** [review]
Didn't see any code issues. Mucked around with the UX a bunch. Found some minor stuff, but nothing that is a blocker for merging this.

**Review by mdbitz — APPROVED** [review]
:wave: Overall this looks good to me. The only question I have is if we need to do any validation/whitelisting of the locale prefix. Is there concern for bad values "https://.jetpack.com" "https://werer.jetpack.com/search", or etc

**Comment by mdbitz** [comment]
@gibrown @jsnmoon 👍 

merging this for 8.3. I'd appreciate your thoughts on hardening the locale prefix (if needed) in a future update.

File: modules/search/instant-search/components/jetpack-colophon.jsx [hunk 1]
```diff
@@ -34,11 +34,16 @@ const logoPath = (
 );
 const logoSize = 12;
 
-const JetpackColophon = () => {
+const JetpackColophon = props => {
+	const locale_prefix = typeof props.locale === 'string' ? props.locale.split( '-', 1 )[ 0 ] : null;
+	const url =
+		locale_prefix && locale_prefix !== 'en'
+			? 'https://' + locale_prefix + '.jetpack.com/search'
```
[the remark this answers is not in the document]
Comment (gibrown) [reply modules/search/instant-search/components/jetpack-colophon.jsx]: I guess in theory the locale for the site could get set to something weird though I am not sure how.

I opened https://github.com/Automattic/jetpack/issues/14785

**Pull request merged**